xdsl/utils/parse_pipeline.py

```diff
@@ -92,3 +93,175 @@ def peek(self) -> Token:
         if self._peeked is None:
             self._peeked = next(self._stream)
         return self._peeked
+
+
+_PassArgTypes = list[str | int | bool | float]
+
+
+@dataclass(eq=True, frozen=True)
+class PipelinePassSpec:
+    """
+    Small dataclass that holds the name and args of a specific pass.
```

Comment:
```suggestion
    A pass name and its arguments.
```
I wouldn't say in documentation that a class is small, or that it is a dataclass, what do you think?